pkg/controllers/termination/terminate.go

```diff
@@ -102,34 +95,30 @@ func (t *Terminator) terminate(ctx context.Context, node *v1.Node) error {
 
 // getPods returns a list of pods scheduled to a node based on some filters
 func (t *Terminator) getPods(ctx context.Context, node *v1.Node) ([]*v1.Pod, error) {
-	pods := &v1.PodList{}
-	if err := t.KubeClient.List(ctx, pods, client.MatchingFields{"spec.nodeName": node.Name}); err != nil {
+	podList := &v1.PodList{}
+	if err := t.KubeClient.List(ctx, podList, client.MatchingFields{"spec.nodeName": node.Name}); err != nil {
 		return nil, fmt.Errorf("listing pods on node, %w", err)
 	}
-	return ptr.PodListToSlice(pods), nil
-}
-
-func (t *Terminator) getEvictablePods(pods []*v1.Pod) []*v1.Pod {
-	evictable := []*v1.Pod{}
-	for _, p := range pods {
+	pods := []*v1.Pod{}
+	for i := range podList.Items {
```

Comment:
Could you do the following? It's not a problem here, but [has been in the past](https://medium.com/swlh/golang-tips-why-pointers-to-slices-are-useful-and-how-ignoring-them-can-lead-to-tricky-bugs-cac90f72e77b) so I'd like to reduce the chance of that happening. You can change the later references to be just `i` in this case.
```suggestion
	for _, i := range ptr.PodListToSlice(podList) {
```

Reply:
You have it backwards. `i` is the index, or `for i, pod := range ...`. I'm avoiding the bug by using the index.

Reply:
Resolved offline. Using `_, i` where i represents the pod.